lib/usd/ui/layerEditor/stageSelectorWidget.cpp

```diff
@@ -112,25 +112,40 @@ void StageSelectorWidget::selectedIndexChanged(int index)
 void StageSelectorWidget::sessionStageChanged()
 {
     if (!_internalChange) {
-        auto index = _dropDown->findData(QVariant::fromValue(_sessionState->stage()));
+        size_t index = -1;
+        for (size_t i = 0; i < _dropDown->count(); i++) {
```

Comment:
This seems silly, ideally I should just do _dropdown->findData, but that doesn't work for some reason, I'm not a Qt expert, but i searched online and with structs it "should" be ok, but it doesn't seem to be.

Reply:
hello. What makes the find work for the UsdStage is the macro at line 17 of this file
Q_DECLARE_METATYPE(pxr::UsdStageRefPtr);

you simply need to change that for the a StageEntry pointer instead

Reply:
sorry, I see you already tried that